### PR TITLE
Add validation of IP addresses and domains in settings

### PR DESCRIPTION
**Pull request by m4xxx1m**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `std::string getHost(const std::string & key) const;` method to "base/poco/Util/include/Poco/Util/AbstractConfiguration.h" which does the same as `Poco::Util::AbstractConfiguration::getString` method but  additionally does validity check to make sure value is a correct IP address or domain name.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

- Motivation: Many settings in ClickHouse's configuration file specify hosts in the form of IP addresses (12.13.14.15) or domains (mail.stanford.edu). Many similar settings exist. As of now, these settings are parsed as strings and typos cannot be detected immediately - in this case the connection to the host would silently fail.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by m4xxx1m** [comment]
#46918

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 6fd7656aebd0d747ddd06a069139cb935031492e with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69130/6fd7656aebd0d747ddd06a069139cb935031492e/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/b6ea3442b420e63a15d847418ec0407dd1009e5a/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by m4xxx1m** [comment]
@nikitamikhaylov As I understand tests fail because they expect docs to be added. I would like to add them, but unfortunately I couldn't found where are POCO libraries docs meant to be placed. Could you please point out what should I do in this situation?

File: base/poco/Util/src/AbstractConfiguration.cpp [hunk 1]
```diff
@@ -529,4 +564,122 @@ void AbstractConfiguration::setRawWithEvent(const std::string& key, std::string
...
+	if (!isValidIPv4Address(value) && !isValidIPv6Address(value) && !isValidDomainName(value))
+	{
+		throw SyntaxException("Property is not a valid host name", value);
+	}
+}
+
+
+bool AbstractConfiguration::isValidIPv4Address(const std::string& value)
+{
+	int octet = 0;
```
Comment (thevar1able) [inline base/poco/Util/src/AbstractConfiguration.cpp:578]: Can we use https://github.com/ClickHouse/ClickHouse/blob/ca514c655687946a9d762252b1e30049c560981e/base/poco/Net/src/IPAddressImpl.cpp#L256 and https://github.com/ClickHouse/ClickHouse/blob/ca514c655687946a9d762252b1e30049c560981e/base/poco/Net/src/IPAddressImpl.cpp#L609?

Comment (m4xxx1m) [reply base/poco/Util/src/AbstractConfiguration.cpp]: Yes, I think it's a good idea. Thank you!

**Review by thevar1able — APPROVED** [review]

**Comment by Algunenano** [comment]
I don't see this new code being used in the project itself? Am I wrong? If it's not used, what's the point of it (we should be removing Poco code, not adding it). Also, it can't be neither a feature or an improvement.

Maybe it's used somewhere and I'm missing it?

**Pull request merged**